### PR TITLE
azurerm_nginx_certificate - add new read only fields to data source

### PR DESCRIPTION
**Pull request by valyria257**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

Since the [2023-09-01 version of the NGINXaaS API](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/nginx/resource-manager/NGINX.NGINXPLUS/stable/2023-09-01/swagger.json#L905), there are new read-only certificate fields: `sha1Thumbprint`, `keyVaultSecretVersion`, `keyVaultSecretCreated`, and `certificateError`. This change adds those fields as attributes on the `azurerm_nginx_certificate` data source.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* azurerm_nginx_certificate - add new read only fields to data source

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: website/docs/d/nginx_certificate.html.markdown [hunk 1]
```diff
@@ -43,6 +43,16 @@ In addition to the Arguments listed above - the following Attributes are exporte
...
 
+* `sha1_thumbprint` - The SHA-1 thumbprint of the certificate.
+
+* `key_vault_secret_version` - The version of the certificate.
+
+* `key_vault_secret_created` - The date/time the certificate was created in Azure Key Vault.
+
+* `certificate_error_code` - The error code of the certificate error, if any.
+
+* `certificate_error_message` - The error message of the certificate error, if any.
```
Comment (katbyte) [inline website/docs/d/nginx_certificate.html.markdown:54]: certificate is aprt of the resource name so these should be:
```suggestion
* `error_code` - The error code of the certificate error, if any.

* `error_message` - The error message of the certificate error, if any.
```

File: website/docs/d/nginx_certificate.html.markdown [hunk 2]
```diff
@@ -43,6 +43,16 @@ In addition to the Arguments listed above - the following Attributes are exporte
 
 * `key_vault_secret_id` - The ID of the Key Vault Secret for the certificate.
 
+* `sha1_thumbprint` - The SHA-1 thumbprint of the certificate.
+
+* `key_vault_secret_version` - The version of the certificate.
+
+* `key_vault_secret_created` - The date/time the certificate was created in Azure Key Vault.
```
Comment (katbyte) [inline website/docs/d/nginx_certificate.html.markdown:50]: would this be more clear as
```suggestion
* `key_vault_secret_creation_date` - The date/time the certificate was created in Azure Key Vault.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the pr @valyria257 - just a few minor comments but otherwise this is looking good! once they have been addressed this should be good to merge

**Comment by valyria257** [comment]
@katbyte thanks for the review! I applied your suggestions :)

**Review by stephybun — APPROVED** [review]
Thanks @valyria257 LGTM 🍀

**Pull request merged**